From c1122e2c2406cbed14bf37e1c6d5bf875fbdf375 Mon Sep 17 00:00:00 2001
From: "Michael T. Kloos" <michael@michaelkloos.com>
Date: Fri, 8 Apr 2022 08:28:42 -0400
Subject: [PATCH] Adjusted the linker flags for kconfig mconf (menuconfig)

For some reason, this breaks on my Gentoo systems.  There,
libcurses does not contain nodelay, so the linking fails
with symbol resolution errors regarding the nodelay symbol.

I have changed it to use pkg-config --libs ncurses, which,
in my case, adds -ltinfo to fix this issue.

Signed-off-by: Michael T. Kloos <michael@michaelkloos.com>
---
 kconfig/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kconfig/Makefile b/kconfig/Makefile
index 2e8e5dcd7..083026d37 100644
--- a/kconfig/Makefile
+++ b/kconfig/Makefile
@@ -67,7 +67,7 @@ gen_config.h: .config
 
 kconfig/mconf: $(SHIPPED)
 	$(HOSTCC) -o $@ kconfig/mconf.c kconfig/zconf.tab.c \
-		kconfig/lxdialog/*.c -lcurses -DCURSES_LOC="<ncurses.h>" \
+		kconfig/lxdialog/*.c `pkg-config --libs ncurses` -DCURSES_LOC="<ncurses.h>" \
 		-DKBUILD_NO_NLS=1 -DPROJECT_NAME=\"$(KCONFIG_PROJECT)\"
 
 kconfig/conf: $(SHIPPED)
